Compilation error in develop branch

**Describe the bug**
Compilation error 

```
/home/amazuyer/dev/geosx/GEOSX/src/components/core/src/managers/BoundaryConditions/BoundaryConditionBase.hpp:658:44: error: cannot call member function ‘void geosx::BoundaryConditionBase::ApplyBoundaryConditionToSystem(geosx::set<long int>&, geosx::real64, geosx::dataRepository::ManagedGroup*, const globalIndex_array&, const integer&, geosx::systemSolverInterface::EpetraBlockSystem*, geosx::systemSolverInterface::BlockIDs, LAMBDA&&) const [with BC_OP = geosx::BcEqual; LAMBDA = geosx::BoundaryConditionBase::ApplyBoundaryConditionToSystem(geosx::set<long int>&, geosx::real64, geosx::dataRepository::ManagedGroup*, const string&, const string&, const integer&, geosx::systemSolverInterface::EpetraBlockSystem*, geosx::systemSolverInterface::BlockIDs) const::<lambda(auto:5)> [with auto:5 = multidimensionalArray::ManagedArray<int, 1, long int>; BC_OP = geosx::BcEqual]::<lambda(geosx::localIndex)>; geosx::set<long int> = SortedArray<long int, long int>; geosx::real64 = double; geosx::globalIndex_array = multidimensionalArray::ManagedArray<long long int, 1, long int>; geosx::integer = int]’ without object
       ApplyBoundaryConditionToSystem<BC_OP>( targetSet,
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
                                              time,
                                              ~~~~~
                                              dataGroup,
                                              ~~~~~~~~~~
                                              dofMap,
                                              ~~~~~~~
                                              dofDim,
                                              ~~~~~~~
                                              blockSystem,
                                              ~~~~~~~~~~~~
                                              blockID,
                                              ~~~~~~~~
                                              [&]( localIndex const a )->real64
                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       {
       ~                                     
         return static_cast<real64>(rtTypes::value( field[a], component ));
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       });
       ~~  
```

**To Reproduce**
Clone develop branch try to configure/compile it with gcc 6.3

**Expected behavior**
Should compile

--- a/src/components/core/src/managers/BoundaryConditions/BoundaryConditionBase.hpp
+++ b/src/components/core/src/managers/BoundaryConditions/BoundaryConditionBase.hpp
@@ -656,13 +656,13 @@ void BoundaryConditionBase::ApplyBoundaryConditionToSystem( set<localIndex> cons
       dataRepository::view_rtype<fieldType> field = view.data();
 
       this->ApplyBoundaryConditionToSystem<BC_OP>( targetSet,
-                                             time,
-                                             dataGroup,
-                                             dofMap,
-                                             dofDim,
-                                             blockSystem,
-                                             blockID,
-                                             [&]( localIndex const a )->real64
+                                                   time,
+                                                   dataGroup,
+                                                   dofMap,
+                                                   dofDim,
+                                                   blockSystem,
+                                                   blockID,
+                                                   [&]( localIndex const a )->real64
       {
         return static_cast<real64>(rtTypes::value( field[a], component ));
       });

